resallocserver/manager.py

```diff
@@ -578,12 +606,23 @@ def _assign_tickets(self):
                     if not ticket_tags.issubset(res_tags):
                         continue
 
-                    # We have found appropriate resource!
+                    priority = 0
+                    for tag in resource.tags:
+                        if tag.priority is not None and tag.id in ticket_tags:
+                            priority += tag.priority
+
+                    queue.add_task(resource, priority)
+
+                # We have found appropriate resource!
+                try:
+                    resource = queue.pop_task()
                     log.debug("Assigning %s to %s", resource.name, ticket.id)
                     assign_ticket(resource, ticket)
                     if ticket.tid:
                         notify_ticket = ticket.tid
-                    break
+                except KeyError:
+                    # empty queue
```

Comment:
The `try:` block is quite wide, I would not be 100% sure that `KeyError` always means empty queue. Maybe it is now but it could change if we do some refactoring and access some dicts in `assign_ticket` or somewhere else. I would do something like

```
try:
    resource = queue.pop_task()
except KeyError:
    # empty queue
    continue  # probably continue if I understand the code properly
```

and then do the rest of the logic with the `resource`.

Reply:
indeed